tests/unit/neptune/new/internal/operation_processors/test_batcher.py

```diff
@@ -0,0 +1,352 @@
+    record2 = QueueElement(LogFloats(["c"], [3.0]), 2, 1)
+
+    # and
+    queue.get.side_effect = (record1, record2, None)
+
+    # and
+    backend.get_int_attribute.return_value = ApiIntAttribute(5)
+
+    # and
+    batcher = Batcher(queue=queue, backend=backend)
+
+    # when
+    result = batcher.collect_batch()
+
+    # then
+    assert result is not None
+
+    # and
+    accumulated_operations, _, version = result
+    assert accumulated_operations.other_operations == [AssignInt(["a"], 5), LogFloats(path=["c"], values=[3.0])]
+    assert accumulated_operations.upload_operations == []
+    assert accumulated_operations.artifact_operations == []
+    assert accumulated_operations.errors == []
+    assert accumulated_operations.source_operations_count == 2
+    assert version == 2
+
+    # and
+    backend.get_int_attribute.assert_called_once_with("123", ContainerType.RUN, ["b"])
+
+
+@patch("neptune.internal.queue.disk_queue.DiskQueue")
+@patch("neptune.internal.backends.factory.HostedNeptuneBackend")
+def test_two_copy_operations_at_begin(backend, queue):
+    # given
+    record1 = QueueElement(CopyAttribute(["a"], "123", ContainerType.RUN, ["b"], IntegerAttribute), 1, 1)
+    record2 = QueueElement(CopyAttribute(["b"], "123", ContainerType.RUN, ["d"], IntegerAttribute), 2, 1)
+    record3 = QueueElement(LogFloats(["c"], [3.0]), 3, 1)
+
+    # and
+    queue.get.side_effect = (record1, record2, record3, None)
+
+    # and
+    backend.get_int_attribute.side_effect = (ApiIntAttribute(5), ApiIntAttribute(3))
+
+    # and
+    batcher = Batcher(queue=queue, backend=backend)
+
+    # when
+    result = batcher.collect_batch()
+
+    # then
+    assert result is not None
+
+    # and
+    accumulated_operations, _, version = result
+    assert accumulated_operations.other_operations == [AssignInt(["a"], 5)]
+    assert accumulated_operations.upload_operations == []
+    assert accumulated_operations.artifact_operations == []
+    assert accumulated_operations.errors == []
+    assert accumulated_operations.source_operations_count == 1
+    assert version == 1
+
+    # and
+    backend.get_int_attribute.assert_called_once_with("123", ContainerType.RUN, ["b"])
+
+
+@patch("neptune.internal.queue.disk_queue.DiskQueue")
+@patch("neptune.internal.backends.factory.HostedNeptuneBackend")
+def test_requires_previous_completed(backend, queue):
+    # given
+    record1 = QueueElement(UploadFileContent(["a"], "flac", "deadbeef"), 1, 1)
+    record2 = QueueElement(DeleteAttribute(["a"]), 2, 1)
+    record3 = QueueElement(UploadFileContent(["a"], "flac", "ffffffff"), 3, 1)
+
+    # and
+    queue.get.side_effect = (record1, record2, record3, None)
+
+    # and
+    batcher = Batcher(queue=queue, backend=backend)
+
+    # when
+    result = batcher.collect_batch()
+
+    # then
+    assert result is not None
+
+    # and
+    accumulated_operations, _, version = result
+    assert accumulated_operations.other_operations == [DeleteAttribute(path=["a"])]
```

Comment:
@BartoszPrusak FYI Pytest properly recognizes `result == expected` :wink: so I'll try to follow this practice.